framework/web/UrlRule.php

```diff
@@ -216,6 +221,145 @@ public function init()
     }
 
     /**
+     * Initializes normalizer configuration
+     * @param UrlManager $manager the URL manager
+     * @throws \Exception if config is not understood
+     */
+    protected function initNormalizer($manager) {
+        $normalize = $this->normalize;
+
+        $configNoAction = [
+            // whether to collapse multiple slashes to single
+            'collapse-slashes' => false,
+            // remove trailing slash if it exists
+            'remove-trailing-slash' => false,
+            // add trailing slash if there is no one
+            'add-trailing-slash' => false,
+            // what action to perform if pathInfo was changed during normalization
+            // redirect: instant redirect to normalized path
+            // route: return new route (must be configured via 'route')
+            // none: no action, just route by normalized path
+            'action' => 'none',
+            // string: route for $action == 'route', i.e. 'site/redirect'
+            'route' => null
+        ];
+
+        if (is_string($normalize) || is_null($normalize)) {
+            switch ($normalize) {
+                case '':
+                case 'safe-remove-trailing-slash':
```

Comment:
It is better to declare such keys as constants - then you do not have to remember them, because the IDE will tell you possible values.